src/main/java/com/google/cloud/spanner/jdbc/CloudSpannerJdbcConnection.java

```diff
@@ -47,6 +50,28 @@
  */
 public interface CloudSpannerJdbcConnection extends Connection {
 
+  /**
+   * Returns the {@link DatabaseId} of the database that this {@link Connection} is connected to.
+   */
+  default DatabaseId getDatabaseId() {
+    throw new UnsupportedOperationException();
+  }
+
+  /**
+   * Returns the underlying {@link DatabaseClient} that is used by this connection. Operations that
+   * are executed on the {@link DatabaseClient} that is returned has no impact on this {@link
+   * Connection}, e.g. starting a read/write transaction on the {@link DatabaseClient} will not
+   * start a transaction on this connection.
+   */
+  default DatabaseClient getDatabaseClient() {
```

Comment:
Can customer do something like below?
```
CloudSpannerJdbcConnection spannerJdbcConnection =
          connection.unwrap(CloudSpannerJdbcConnection.class);
      List<String> ddl =
          spannerJdbcConnection
              .getSpanner()
              .getDatabaseClient()
```
Is there any difference between this and the exposed method? I mean do we need to expose a seperate method for `getDatabaseClient`?

Reply:
At the moment, there is no difference, as the only key that is used to determine which `DatabaseClient` to get from a `Spanner` instance is the `DatabaseId`. But this might change in the future, for example when #1856 is resolved. With this, the JDBC driver might use additional properties to determine which `DatabaseClient` is used for a connection.